client_test.go

```diff
@@ -146,29 +110,10 @@ func TestCanReceiveCLEARCHATMessage(t *testing.T) {
 	testMessage := `@ban-duration=1;ban-reason=testing\sxd;room-id=11148817;target-user-id=40910607 :tmi.twitch.tv CLEARCHAT #pajlada :ampzyh`
 	wait := make(chan struct{})
 
-	go func() {
-		cer, err := tls.LoadX509KeyPair("test_resources/server.crt", "test_resources/server.key")
-		if err != nil {
-			log.Println(err)
-			return
-		}
-		config := &tls.Config{
-			Certificates: []tls.Certificate{cer},
-		}
-		ln, err := tls.Listen("tcp", ":4323", config)
-		if err != nil {
-			t.Fatal(err)
-		}
-		close(wait)
-		conn, err := ln.Accept()
-		if err != nil {
-			t.Fatal(err)
-		}
-		defer ln.Close()
-		defer conn.Close()
-
+	var conn net.Conn
+	go createServer(t, ":4323", &conn, wait, func() {
 		fmt.Fprintf(conn, "%s\r\n", testMessage)
```

Comment:
This is not how the server should work. The test should only call the server with a port the logic with what the server replies should all be defined in the mock

Reply:
So the goal would be to simulate the entire Twitch API server?

Reply:
well not in it's entirety, but the few simple things that are easily done like sending PINGs, 001 messages, etc.